2019/05/07/simulate.cpp

```diff
@@ -83,17 +85,33 @@ void demo(size_t N, uint64_t range_min, uint64_t range_max) {
     printf("Generating %zu values in [%zu, %zu), density: %f %%\n", N, range_min, range_max, N*100.0 / (range_max-range_min));
     uint64_t * array = (uint64_t *) malloc(sizeof(uint64_t) * N);
     if(array == NULL) return;
-     clock_t bef1 = clock();
-    size_t c1 = pick_N(range_min, range_max, N, array);
-         clock_t aft1 = clock();
-     clock_t bef2 = clock();
 
-    size_t c2 = fast_pick_N(range_min, range_max, N, array);
-         clock_t aft2 = clock();
-    printf("timings: %f s %f s \n",(double)(aft1-bef1)/CLOCKS_PER_SEC, (double)(aft2-bef2)/CLOCKS_PER_SEC);
-    printf("timings per value: %f ns %f ns \n",(double)(aft1-bef1 )* 1000  * 1000  * 1000/(N * CLOCKS_PER_SEC), (double)(aft2-bef2)* 1000 * 1000 * 1000/(N * CLOCKS_PER_SEC));
+    clock_t bef1 = clock();
+    size_t c1 = pick_N(range_min, range_max, N, array, [](uint64_t *array, size_t n) {
+      qsort(array, n, sizeof(uint64_t), uint64_comparison);
+    });
+    clock_t aft1 = clock();
 
-    printf("actual counts: %zu %zu \n",c1, c2);
+    clock_t bef2 = clock();
+    size_t c2 = pick_N(range_min, range_max, N, array, [](uint64_t *array, size_t n) {
+      std::sort(array, array + n);
```

Comment:
most of this is just reformatting: these three lines are the one ones alogn with updated output